includes/abstracts/abstract-wc-order.php

```diff
@@ -932,6 +932,44 @@ public function add_discount( $discount ) {
 	}
 
 	/**
+	 * Remove a coupon from the order and recalculate totals.
+	 *
+	 * @since 3.2
+	 * @param string $code Coupon code
+	 * @return void
+	 */
+	public function remove_coupon( $code ) {
```

Comment:
This approach seems wasteful because it rebuilds all of the coupon and discount order items, but in theory it should always work without having to figure out a whole system for reversing the discounts from the subtotals.